### PR TITLE
Closes #176: Added Role Delegation module and permissions.

### LINKED ISSUE #176: Role Delegation
## User Story(s)
As a _content administrator_ on a Quickstart site, I'd like to be able to add other _content admins_ and _content editors_ to my site. 
…
## Conditions of satisfaction
- [ ] Should provide way for users with the `az_content_admin` role to add additional users to the site and grant the `az_content_editor` and `az_content_admin` role
- [ ] Should not allow users without the `administrator` role to perform to grant the `administrator` role to other users
- [ ] Should not allow users with the `az_content_admin` role to change user account settings (including which role is the site administrator role)

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
Adds Role delegation module and permissions.

## Related Issue
#176 

## How Has This Been Tested?

Locally with lando.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by joeparsons** [comment]
This probably needs a DB update to enable the module on existing sites and possibly to update the existing role permissions...

**Comment by camikazegreen** [comment]
Per today's discussion, these permissions were going to be added to a separate user role: User Admin

I would suggest adding a new file: user.role.az_user_admin.yml
`
langcode: en
status: true
dependencies: {  }
id: az_user_admin
label: 'User administrator'
weight: 4
is_admin: false
permissions:
  - 'administer users'
  - 'assign az_content_admin role'
  - 'assign az_content_editor role'`

**Comment by joeparsons** [comment]
> Per today's discussion, these permissions were going to be added to a separate user role: User Admin

OK.  Could we call it `user manager` maybe?

**Comment by camikazegreen** [comment]
I don't actually know if we had much discussion about the actual role name, I thought I saw you suggest User Manager, which sounds great to me.

**Comment by kevcooper** [comment]
Tested with lando: a user with the "User administrator" role was able to log in and add the "Content administrator" role to an account that previously only had the "Content editor" role applied.

Seems to work as expected!

**Review by kevcooper — APPROVED** [review]

**Comment by trackleft** [comment]
Same as @kevcooper 

Logged in with each different role and was only able to alter roles with the administrator and user admin roles.

Here are the options I saw with the user admin permissions.

```<select data-drupal-selector="edit-action" id="edit-action" name="action" class="form-select form-element form-element--type-select" style="display:visible; position:absolute; width:0px; height: 0px; clip:rect(0,0,0,0)" tabindex="-1">
	<option value="pathauto_update_alias_user">Update URL alias</option>
	<option value="user_add_role_action.az_content_admin">Add the Content administrator role to the selected user(s)</option>
	<option value="user_add_role_action.az_content_editor">Add the Content editor role to the selected user(s)</option>
	<option value="user_block_user_action">Block the selected user(s)</option>
	<option value="user_cancel_user_action">Cancel the selected user account(s)</option>
	<option value="user_remove_role_action.az_content_admin">Remove the Content administrator role from the selected user(s)</option>
	<option value="user_remove_role_action.az_content_editor">Remove the Content editor role from the selected user(s)</option>
	<option value="user_unblock_user_action">Unblock the selected user(s)</option>
</select>```

**Review by trackleft — APPROVED** [review]

File: config/install/user.role.az_user_admin.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+status: true
+dependencies: {  }
+id: az_user_admin
+label: 'User administrator'
+weight: 4
+is_admin: false
+permissions:
+  - 'administer users'
+  - 'assign az_content_admin role'
+  - 'assign az_content_editor role'
```
Comment (camikazegreen) [inline config/install/user.role.az_user_admin.yml:11]: Do we need to grant this permission also?
```suggestion
  - 'assign az_user_admin role'
```

Comment (camikazegreen) [inline config/install/user.role.az_user_admin.yml:11]: Do we need to add this role as well?
```suggestion
  - 'assign az_content_editor role'
  - 'assign az_user_admin role'
```

**Review by kevcooper — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Review by jwmoore1 — APPROVED** [review]

**Review by camikazegreen — APPROVED** [review]

**Pull request merged**